workflow/build_compute.py

```diff
@@ -0,0 +1,176 @@
+    """
+    Generate a task specification dictionary for a given task.
+
+    Parameters
+    ----------
+    task_name: str
+        The name of the task.
+    task_spec: Dict
+        The specification of the task, containing command, walltime, and cores.
+    host_spec: Dict
+        The specification of the host, containing account, queue, partition, and native.
+
+    Returns:
+    --------
+    task_dict: Dict
+        A dictionary containing the task specification, including resources and other task-related information.
+    """
+
+    task_dict = AttrDict()
+    task_dict.task_name = task_name
+    task_dict.cycledef = "build"
+    task_dict.maxtries = 1
+    task_dict.command = f"cd {HOMEgfs}/sorc/; {task_spec.command}"
+    task_dict.job_name = task_name
+    task_dict.log = f"{HOMEgfs}/sorc/logs/{task_name}.log"
+
+    task_dict.resources = AttrDict()
+    task_dict.resources.account = host_spec.account
+    task_dict.resources.queue = host_spec.queue
+    task_dict.resources.partition = host_spec.partition
+    task_dict.resources.walltime = task_spec.walltime
+    task_dict.resources.native = host_spec.native
+    task_dict.resources.memory = None
+    task_dict.resources.nodes = 1
+    task_dict.resources.ntasks = task_spec.cores
+    task_dict.resources.ppn = task_spec.cores
+    task_dict.resources.threads = 1
+
+    return task_dict
+
+
+def get_host_specs(host: Dict) -> Dict:
+    """Generate host specs for the build.xml file based on Host() info
+
+    Parameters
+    ----------
+    host : Dict
+        Host information returned by Host()
+
+    Returns
+    -------
+    specs: Dict
+        Consolidated compute specifics needed for the XML
+    """
+
+    native = None
+    partition = None
+
+    if host.info.SCHEDULER in ['pbspro']:
+        native = '-l place=vscatter'
+    elif host.info.SCHEDULER in ['slurm']:
+        native = '--export=NONE'
+        if host.info.PARTITION_BATCH not in [""]:
+            partition = host.info.PARTITION_BATCH
+
+    if host.info.RESERVATION not in [""]:
+        native += f' --reservation={host.info.RESERVATION}'
+
+    if host.info.CLUSTERS not in [""]:
+        native += f' --clusters={host.info.CLUSTERS}'
+
+    specs = AttrDict()
+    specs.scheduler = host.info.SCHEDULER
+    specs.account = host.info.ACCOUNT
+    specs.queue = host.info.QUEUE
+    specs.partition = partition
+    specs.native = native
+
+    return specs
+
+
+def main(*argv):
+
+    user_inputs = input_args(*argv)
+    host_specs = get_host_specs(Host())
+
+    # Update the default host account if the user supplied one
+    if user_inputs.account is not None:
+        host_specs.account = user_inputs.account
+
+    build_specs = AttrDict(parse_yaml(user_inputs.yaml))
+
+    # Determine systems to build
+    builds = set()
+    if user_inputs.system == "all":
+        for system, build_set in build_specs.systems.items():
+            builds.update(build_set)
```

Comment:
Isn't this just the `build_specs.build` instead of going through systems?

Reply:
Ah, yes, we only have to loop through systems if something besides `all` is selected.